tests/framework/cli/conftest.py

```diff
@@ -33,13 +33,12 @@
 
 @fixture
 def entry_points(mocker):
-    return mocker.patch("importlib_metadata.entry_points")
+    return mocker.patch("importlib_metadata.entry_points", spec=True)
 
 
 @fixture
 def entry_point(mocker, entry_points):
-    ep = mocker.MagicMock()
-    entry_points.return_value = ep
+    ep = mocker.patch("importlib_metadata.EntryPoint", spec=True)
```

Comment:
`spec=True` is needed to trigger `AttributeError` when accessing non-exist attribute. I wonder if we may need to update other tests eventually. 

But I haven't fully understood these parameters, so will do it later instead. I have raise an issue on pytest repo for this.

Reply:
This doesn't look right to me 🤔  I don't think we should need `spec=True` here. I think we're probably just mocking the wrong things or don't have the right tests to pick up on the bug.

Reply:
Did you manage to find out if `spec=True` ok here?

Reply:
Did you manage to find out if `spec=True` ok here?

Reply:
@SajidAlamQB  I think using `spec` which mock the real object is better than using `MagicMock`.

![image](https://user-images.githubusercontent.com/18221871/170984424-98c672e8-f7d3-4bcd-a098-b02142fb4dc7.png)

Reply:
Looks like we'd need to set `spec=True` for both Mock and MagicMock: https://stackoverflow.com/questions/52978673/how-does-magicmock-avoid-throwing-attributeerror-when-a-random-method-is-called

Curious as to how this is handled in other instances without using spec 🤔

Reply:
Here is a mistake that I made initially. The mock only cares about the method/class that it mocks, not the return value.

For example, if we have a function `create_kedrocontext(args) -> KedroContext`. If we mock `create_kedrocontext` with `spec=True`, it will return a generic `MagicMock`.
 If we want the return value to respect `KedroContext`'s spec, we need to specify the return value as `MagicMock(KedroContext`.